src/sync/sync_session.hpp

```diff
@@ -177,6 +197,10 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
 
     bool can_wait_for_network_completion() const;
 
+    // Whether or not the session object in `m_session` has been `bind()`ed before.
+    // This determines how the `SyncSession` behaves when refreshing tokens.
+    bool m_session_has_been_bound;
```

Comment:
This member should be declared down near `m_session`, rather than amongst unrelated functions.